processor/batchprocessor/factory.go

```diff
@@ -30,6 +30,11 @@ const (
 
 	defaultSendBatchSize = uint32(8192)
 	defaultTimeout       = 200 * time.Millisecond
+
+	// defaultMetadataCardinalityLimit should be set to the number
+	// of metadata configurations the user expects to submit to
+	// the collector.
+	defaultMetadataCardinalityLimit = 1000
```

Comment:
There was a discussion in the SIG call yesterday about the defaults for the batch processor being too high, i wonder if we should start with a lower limit than this and increase it if users find it to be too low.

Reply:
I raised it from 100 to 1000 based on the comment here: https://github.com/open-telemetry/opentelemetry-collector/pull/7325#discussion_r1161766124